src/metabase/notification/payload/impl/system_event.clj

```diff
@@ -126,6 +128,25 @@
                             :status "discontinued"}}
      :map]]])
 
+(defn- normalize-record
+  [record field-names]
+  ;; handle different casing from different driver
+  (let [field-names      (set field-names)
+        col->field-name  (merge (u/for-map [f field-names]
+                                  [(u/lower-case-en f) f])
+                                (u/for-map [f field-names]
+                                  [(u/upper-case-en f) f]))]
```

Comment:
You need to also merge `(zipmap field-names field-names)` to handle case sensitive databases where the name is mixedcase.